app/javascript/packages/document-capture/hooks/use-log-camera-info.ts

```diff
@@ -0,0 +1,69 @@
+import { useState, useEffect, useContext } from 'react';
+import AnalyticsContext from '../context/analytics';
+
+function getConstraints(deviceId) {
+  return {
+    video: {
+      width: {
+        ideal: 999999,
+      },
+      height: {
+        ideal: 999999,
+      },
+      deviceId: {
+        exact: deviceId,
+      },
+    },
+  };
+}
+
+function getCameraInfo(videoTrack) {
+  const cameraInfo = {
+    label: videoTrack.label,
+    frameRate: videoTrack.getSettings().frameRate,
+    height: videoTrack.getSettings().height,
+    width: videoTrack.getSettings().width,
+  };
+  return cameraInfo;
+}
+
+async function updateConstraintsAndLogInfo(videoDevice, trackEvent) {
+  // See https://developer.mozilla.org/en-US/docs/Web/API/MediaTrackConstraints/facingMode
+  const updatedConstraints = getConstraints(videoDevice.deviceId);
+  try {
+    const stream = await navigator.mediaDevices.getUserMedia(updatedConstraints);
+    const videoTracks = stream.getVideoTracks();
+    const cameras = videoTracks.map((videoTrack) => getCameraInfo(videoTrack));
+    const logInfo = {
+      camera_info: cameras,
+    };
+    trackEvent('IdV: camera info logged', logInfo);
+  } catch (error) {
+    trackEvent('IdV: camera info error', error);
```

Comment:
Can we match the event name so that we can add to `FrontendLogController::ALLOWED_EVENTS` rather than the legacy `FrontendLogController::LEGACY_EVENT_MAP` ?

```suggestion
    trackEvent('idv_camera_info_logged', logInfo);
  } catch (error) {
    trackEvent('idv_camera_info_error', error);
```

Reply:
Also, based on the method signature in `analytics_events.rb`, is this expected to pass `error` as a property of an object? Right now it's sending the error directly. It doesn't look like we have any test coverage for this.

Reply:
[would `trackError` possibly work](https://github.com/18F/identity-idp/blob/83d39c443d54bda6f8202b93695f51bca31b4169/app/javascript/packages/analytics/index.ts#L28)?

Reply:
- I [changed the names ](https://github.com/18F/identity-idp/pull/10227/commits/167cc6d61a56a5438f5bbf29b690d08097a2a19a)of both events so they're not in the legacy naming scheme anymore.
- I [removed the error ](https://github.com/18F/identity-idp/pull/10227/commits/5250c0d18372656357c1e402d75ffbb4e110f519)information from the logging. We'll still track that it happened, just not what happened.
- I looked at adding `trackError` here, but it's not included in `AnalyticsContext` and I don't want to do that as part of this PR.